diag_manager/fms_diag_object.F90

```diff
@@ -770,9 +773,22 @@ function fms_diag_do_reduction(this, field_data, diag_field_id, oor_mask, weight
   logical                   :: block_in_subregion !< .True. if the current block is part of the subregion
   integer                   :: starting           !< Starting index of the subregion relative to the compute domain
   integer                   :: ending             !< Ending index of the subregion relative to the compute domain
+  real(kind=r8_kind)        :: missing_value      !< Missing_value for data points that are masked
```

Comment:
This is saved as an r8, passed into the do_time_* subroutines and converted to r4 later:
https://github.com/NOAA-GFDL/FMS/blob/011b414d6fcdab63195c8946eef783a1de77c5d0/diag_manager/fms_diag_output_buffer.F90#L497

This is to avoid having even more (2) select types.
one here: 
https://github.com/NOAA-GFDL/FMS/blob/011b414d6fcdab63195c8946eef783a1de77c5d0/diag_manager/fms_diag_object.F90#L783-L789

and another one here:
https://github.com/NOAA-GFDL/FMS/blob/011b414d6fcdab63195c8946eef783a1de77c5d0/diag_manager/fms_diag_output_buffer.F90#L487-L501